File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/SetCMPIdBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton

internal class SetCMPIdBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "setCMPId"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        usercentrics.instance.setCMPId(call.arguments as Int)
        result.success(null)
    }

}

File: ios/Classes/Bridge/SetCMPIdBridge.swift
import Usercentrics

struct SetCMPIdBridge : MethodBridge {

    let name: String = "setCMPId"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        usercentrics.shared.setCMPId(id: call.arguments as! Int32)
        result(nil)
    }
}
